assembly/src/tokens/mod.rs

```diff
@@ -247,7 +250,8 @@ fn validate_proc_declaration_label(label: &str, token: &Token) -> Result<String,
         return Err(ParsingError::invalid_proc_label(token, label));
     }
 
-    Ok(label.to_string())
+    ProcedureName::try_from(label.to_string())
+        .map_err(|e| ParsingError::invalid_library_path(token, e))
```

Comment:
Why not move the above checks into `ProcedureName::try_form()`? Seems like it might be a good single place for procedure name validation.

Reply:
I initially started that, but then I got the feeling I was mixing parsing logic with library path validation. Maybe they should be two different domains. Wdyt?

Reply:
I would combine them. I think it will make things a bit cleaner.

Reply:
btw, as I was working on #381 I realized that we may need to make the error a bit more general as invalid procedure name could be either `ParsingError` or `LibraryError`. So, maybe we should have a separate error type to deal with invalid procedure names, module names, paths etc. Maybe it should be something like `PathError` (though I don't love it).

Also, I realized that we are missing quite a few checks in the current procedure name validation. I've added the updated logic [here](https://github.com/maticnetwork/miden/blob/6fbac4b3ea1ebd8f6fcaf1a53a73c80acfdf6578/assembly/src/tokens/mod.rs#L244), but to summarize:
- Procedure name must start with a letter.
- It can consist only of letters, numbers, and underscores.
- It cannot be more than 100 characters long (`MAX_PROC_NAME_LEN`).

Reply:
I was thinking of introducing a new error type which would not require a token to be constructed - but yeah, let's do it in a separate PR. Could you create an issue for this?

Reply:
Issue added: https://github.com/maticnetwork/miden/issues/583